### PR TITLE
feat: update to use Batch value get instead of List and Fetch all secrets for AWS provider

### PR DESCRIPTION
**Pull request by Skarlso**
## Problem Statement

What is the problem you're trying to solve?

## Related Issue

Fixes https://github.com/external-secrets/external-secrets/issues/2902

## Proposed Changes

How do you like to solve the issue and why?

## Checklist

- [ ] I have read the [contribution guidelines](https://external-secrets.io/latest/contributing/process/#submitting-a-pull-request)
- [ ] All commits are signed with `git commit --signoff`
- [ ] My changes have reasonable test coverage
- [ ] All tests pass with `make test`
- [ ] I ensured my PR is ready for review with `make reviewable`

### CODE REVIEW

**Comment by Skarlso** [comment]
I expect the AWS e2e test to fail because of the missing permission.

**Comment by Skarlso** [comment]
This is a breaking change because of the permission. But making it backwards compatible would result in people not being aware of the change.

File: pkg/provider/aws/secretsmanager/secretsmanager.go [hunk 1]
```diff
@@ -410,19 +412,21 @@ func (sm *SecretsManager) findByTags(ctx context.Context, ref esv1beta1.External
 	var nextToken *string
 	for {
 		log.V(1).Info("aws sm findByTag", "nextToken", nextToken)
-		it, err := sm.client.ListSecrets(&awssm.ListSecretsInput{
+		it, err := sm.client.BatchGetSecretValueWithContext(ctx, &awssm.BatchGetSecretValueInput{
```
Comment (gusfcarvalho) [inline pkg/provider/aws/secretsmanager/secretsmanager.go:415]: what happens if we have a `find` operation that would target more than 20 Secrets ? 😄 Shouldn't we perform any additional logic (I would expecting at least like an `Iter` object or a `Paginator` in order to continue the search?

Comment (Skarlso) [reply pkg/provider/aws/secretsmanager/secretsmanager.go:415]: But this is in a for loop already that paginates with next token.

Comment (gusfcarvalho) [reply pkg/provider/aws/secretsmanager/secretsmanager.go:415]: yup. Too soon on a saturday for accurate reviews 😄

**Review by gusfcarvalho — APPROVED** [review]
We just need to proper communicate that on a release note, but IMO this is so specific to AWS i'm not sure its worth the minor bump 🤔 

(and fix the e2e)

**Comment by Skarlso** [comment]
@gusfcarvalho do you know where I can update the e2e aws permissions? 🤣

**Comment by Skarlso** [comment]
hum. Tried it with the following:

```yaml
apiVersion: external-secrets.io/v1beta1
kind: ExternalSecret
metadata:
  name: example
spec:
  refreshInterval: 1m
  secretStoreRef:
    name: aws-secretsmanager
    kind: SecretStore
  target:
    name: aws-test-secret
    creationPolicy: Owner
  dataFrom:
  - find:
      path: dev
      name: 
        regexp: "dev/stage1/secret-*"

```

worked

```
FOUND SECRET:  {
  ARN: "arn:aws:secretsmanager:us-west-2:<id>:secret:dev/stage1/secret-name-2-AN5mny",
  CreatedDate: 2024-12-08 18:11:19.093 +0000 UTC,
  Name: "dev/stage1/secret-name-2",
  SecretString: <sensitive>,
  VersionId: "47075aad-f16f-4cf8-8309-5dcf80ece71c",
  VersionStages: ["AWSCURRENT"]
}
```

not sure what's wrong with the e2e test.

**Comment by Skarlso** [comment]
Works with multiple secrets as well.

```
data:
  dev_stage1_secret-name-2: eyJ0ZXN0Ijoic2VjcmV0In0=
  dev_stage1_secret-name-3: eyJ0ZXN0MiI6InRlc3QifQ==

```

**Comment by Skarlso** [comment]
@gusfcarvalho can I get a re-approve please? :)))

**Review by gusfcarvalho — APPROVED** [review]

**Pull request merged**